src/client.ts

```diff
@@ -658,9 +686,7 @@ export class StreamChat<
     this.tokenManager.reset();
 
     // close the WS connection
-    if (this.wsConnection) {
-      return this.wsConnection.disconnect(timeout);
-    }
+    this.disconnectUser(timeout);
```

Comment:
this needs to be run on the function entry point before we reset the activeChannels state